inference-engine/src/transformations/src/transformations/common_optimizations/split_squeeze_concat_fusion.cpp

```diff
@@ -0,0 +1,80 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "itt.hpp"
+#include "transformations/common_optimizations/split_squeeze_concat_fusion.hpp"
+
+#include <memory>
+#include <vector>
+#include <numeric>
+
+#include <ngraph/opsets/opset7.hpp>
+#include <ngraph/rt_info.hpp>
+#include <ngraph/pattern/op/wrap_type.hpp>
+
+NGRAPH_RTTI_DEFINITION(ngraph::pass::SplitSqueezeConcatFusion, "SplitSqueezeConcatFusion", 0);
+
+ngraph::pass::SplitSqueezeConcatFusion::SplitSqueezeConcatFusion() {
+    MATCHER_SCOPE(SplitSqueezeConcatFusion);
+    // Detect only concat, because we don't know how many inputs will go into concat
+    auto concat = ngraph::pattern::wrap_type<ngraph::opset7::Concat>();
+
+    ngraph::matcher_pass_callback matcher_pass_callback = [=](ngraph::pattern::Matcher& m) {
+        auto& pattern_to_output = m.get_pattern_value_map();
+        auto concat_node = std::dynamic_pointer_cast<ngraph::opset7::Concat>(pattern_to_output.at(concat).get_node_shared_ptr());
+        if (!concat_node)
+            return false;
+        auto squeeze0 = std::dynamic_pointer_cast<ngraph::opset7::Squeeze>(concat_node->input_value(0).get_node_shared_ptr());
+        if (!squeeze0)
+            return false;
+        auto split_node = std::dynamic_pointer_cast<ngraph::opset7::Split>(squeeze0->input_value(0).get_node_shared_ptr());
+        if (!split_node)
+            return false;
+        auto axis_node = std::dynamic_pointer_cast<ngraph::opset7::Constant>(split_node->input_value(1).get_node_shared_ptr());
+        if (!axis_node)
+            return false;
+        auto axis_vec = axis_node->cast_vector<int64_t>();
+        if (axis_vec.size() != 1)
+            return false;
+        auto axis_value = axis_vec[0];
+        ngraph::NodeVector nodes_to_delete{ split_node, concat_node };
+        for (auto& inp : concat_node->inputs()) {
+            auto squeeze_node = std::dynamic_pointer_cast<ngraph::opset7::Squeeze>(inp.get_source_output().get_node_shared_ptr());
+            if (!squeeze_node)
+                return false;
+            nodes_to_delete.push_back(squeeze_node);
+            auto squeeze_axes = std::dynamic_pointer_cast<ngraph::opset7::Constant>(squeeze_node->input_value(1).get_node_shared_ptr());
+            if (!squeeze_axes)
+                return false;
+            auto squeeze_axes_vec = squeeze_axes->cast_vector<int64_t>();
+            if (squeeze_axes_vec.size() != 1 || squeeze_axes_vec[0] != axis_value)
+                return false;
+        }
```

Comment:
You also need to check that Squeeze operation are connected to Split in specific order.